analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -307,6 +340,7 @@ def get_checker_list(self, config) -> Tuple[List[str], List[str]]:
                                 "instead.")
                     if state == CheckerState.enabled:
                         compiler_warnings.add('-W' + warning_name)
+                        enabled_checkers.add(checker_name)
```

Comment:
Can you document why were doing this in code?

Reply:
I added this documentation to the beginning. Would you think, it explains?
```.py
# In clang-tidy the warnings as compiler-diagnostics are enabled and
# disabled by -W<name> and -Wno-<name> respectively. The
# clang-diagnostic-<name> is needed, because by default all checkers
# are disabled by "checks=-*" and warnings have to be enabled as
# checkers with this name: clang-diagnostic-<name>.
```